### PR TITLE
Issue/843 add autocomplete suggestions

### LINKED ISSUE #843: Search: Add Autocomplete with Suggestions
Add autocomplete suggestion to the search input field as shown in the mockup below.

<a href="https://user-images.githubusercontent.com/3827611/66611860-0f33c080-eb7d-11e9-95d7-9c689317bf39.png"></a>

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add autocomplete suggestion to the search input field to close #843.  See the screenshots and animation below for illustration.

843_add_autocomplete_suggestions

<a href="https://user-images.githubusercontent.com/3827611/67818109-a2be2a00-fa75-11e9-9e4e-40cdc7a0f2c5.gif"></a>

### Test
1. Tap ***Search*** action in top app bar.
2. Notice note list is blank.
3. Enter text into search field.
4. Notice suggestions appear based on entered text.
5. Tap any suggestion.
6. Notice suggestion text is inserted into search field.
7. Notice note list is shown based on inserted text.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java [hunk 1]
```diff
@@ -673,19 +691,31 @@ public void setNoteSelected(String selectedNoteID) {
...
         mSortOrder.setText(R.string.sort_search_relevance);
-        refreshListForSearch();
 
         if (!searchString.equals(mSearchString)) {
             mSearchString = searchString;
-            refreshList();
+        }
+
+        if (searchString.isEmpty()) {
+            // TODO: Get history items.
```
Comment (AmandaRiu) [inline Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java]: Just verifying this `// TODO` isn't an oversight

**Comment by AmandaRiu** [comment]
Ready for merge, just waiting on design review.

**Comment by SylvesterWilmott** [comment]
Looks great @theck13 

Can you explain how the tag suggestion order works? I created tags named "sa" "sb" "sc" .. etc and when searching "s" one of my older tags "plants" is top of my list, is this because it's used in more notes?

Also when I type "tag:" or "tag:[x]" in the search bar, I no longer have suggestions.

**Comment by theck13** [comment]
> Can you explain how the tag suggestion order works?

The tags are ordered by their key, which is effectively the created date, in descending order.

> Also when I type "tag:" or "tag:[x]" in the search bar, I no longer have suggestions.

Unless you have note or tag with "tag:" in it, searching for "tag:" will show no results.

> In landscape orientation, I also noticed the "X" button in the search bar is not aligned to the right, is this an artefact from our tablet view?

The search view is part of the top app bar and not either of the note list or note editor/preview panes below it.  The "X" button is part of that view and isn't modified by these changes.

> Can you also add some tablet screens to the PR too?

Before|After
-|-
843_add_autocomplete_suggestions_tablet_before_01|843_add_autocomplete_suggestions_tablet_after_01
843_add_autocomplete_suggestions_tablet_before_02|843_add_autocomplete_suggestions_tablet_after_02
843_add_autocomplete_suggestions_tablet_before_03|843_add_autocomplete_suggestions_tablet_after_03

**Comment by theck13** [comment]
@SylvesterWilmott and I discussed how to handle the tag suggestion sorting and behavior.  We came up with this:

<blockquote>
<ul>
<li>
Tag suggestions are always ordered by most used to least used (i.e. tags assigned to the most notes are listed first/top).
</li>
<li>
All tags are shown when `tag:` is entered.
</li>
</blockquote>

**Comment by theck13** [comment]
<blockquote>
<ul>
<li>
Tag suggestions are always ordered by most used to least used (i.e. tags assigned to the most notes are listed first/top).
</li>
</blockquote>

Updated tag suggestions order in https://github.com/Automattic/simplenote-android/pull/865/commits/78a16141d83709d0cec4f82f6705da3b65f40094.

<blockquote>
<li>
All tags are shown when <code>tag:</code> is entered.
</li>
</blockquote>

Updated showing all tags with `tag:` in https://github.com/Automattic/simplenote-android/pull/865/commits/205711fcbd0ece9cab61e2dd5727fdea1503c996.

**Comment by theck13** [comment]
I updated the behavior when tapping a tag from the suggestions in a40224d1 so that multiple tags can be searched at once.  See the animation below for illustration.

<a href="https://user-images.githubusercontent.com/3827611/68172654-93d3ed80-ff35-11e9-8651-cb03aed6dfe7.gif"></a>

**Comment by SylvesterWilmott** [comment]
Looks good @theck13

**Pull request merged**